scene/resources/resource_format_text.cpp

```diff
@@ -907,6 +907,84 @@ Error ResourceLoaderText::rename_dependencies(Ref<FileAccess> p_f, const String
 	return OK;
 }
 
+// Fetches the script attached to a custom resource, reading as few lines as possible from the resource.
+// Please call ResourceLoaderText::open() before calling this.
+Error ResourceLoaderText::get_resource_script_path(String &r_script_path) {
+	if (res_type != "Resource") {
+		error_text = "Expected a user-defined resource file";
+		_printerr();
+		error = ERR_PARSE_ERROR;
+		return error;
+	}
+
+	while (true) {
+		if (next_tag.name != "ext_resource") {
+			break;
+		}
```

Comment:
I may be misinterpreting this, but judging from the way this loop is designed, it seems to break out with the first `ext_resource` it finds. idk if that's actually guaranteed to refer to the `script` property of the resource though (is it?). Worst case scenario (and what I suspect), you'd have to jump to the serialized properties, find the one named `script`, identify which ID number of `ExtResource(<id>)` is assigned to it, and THEN work backward from there to find which `ext_resource` tag at the top has that ID.